jwst/outlier_detection/outlier_detection_spec.py

```diff
@@ -103,16 +110,17 @@ def do_detection(self):
             log.info("Writing out MEDIAN image to: {}".format(
                      median_model.meta.filename))
             median_model.save(median_model.meta.filename)
+            self.output_list.append(median_model.meta.filename)
 
         if pars['resample_data'] is True:
             # Blot the median image back to recreate each input image specified
             # in the original input list/ASN/ModelContainer
             blot_models = self.blot_median(median_model)
             if save_intermediate_results:
                 log.info("Writing out BLOT images...")
-                blot_models.save(
-                    partial(self.make_output_path, suffix='blot')
-                )
+                blot_file = partial(self.make_output_path, suffix='blot')
+                blot_models.save(blot_file)
+                self.output_list.append(blot_file)
```

Comment:
Since this code is within a `if save_intermediate_results` block, is it necessary to add the item to the `output_list` (which appears to only be used if `save_intermediate_results` is `False`)?

Reply:
I fixed the code to do the cleaning after the files are no longer needed instead of passing the list out.